src/overlays/actors/ovl_Boss_Fd2/z_boss_fd2.c

```diff
@@ -22,183 +59,1157 @@ const ActorInit Boss_Fd2_InitVars = {
+                damage = (hurtbox->toucher.dmgFlags & 0x00001000) ? 4 : 2;
+            } else {
+                canKill = true;
+            }
+            if (hurtbox->toucher.dmgFlags & 0x80) {
+                damage = 0;
+            }
+            if (((s8)bossFd->actor.colChkInfo.health > 2) || canKill) {
+                bossFd->actor.colChkInfo.health -= damage;
+                osSyncPrintf(VT_FGCOL(GREEN));
+                osSyncPrintf("damage   %d\n", damage);
+            }
+            osSyncPrintf(VT_RST);
+            osSyncPrintf("hp %d\n", bossFd->actor.colChkInfo.health);
+
+            if ((s8)bossFd->actor.colChkInfo.health <= 0) {
+                bossFd->actor.colChkInfo.health = 0;
+                BossFd2_SetupDeath(this, globalCtx);
+                this->work[FD2_DAMAGE_FLASH_TIMER] = 10;
+                this->work[FD2_INVINC_TIMER] = 30000;
+                Audio_SetBGM(0x100100FF);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_DEAD);
+                func_80032C7C(globalCtx, &this->actor);
+            } else if (damage) {
+                BossFd2_SetupDamaged(this, globalCtx);
+                this->work[FD2_DAMAGE_FLASH_TIMER] = 10;
+                this->work[FD2_INVINC_TIMER] = 100;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_VALVAISA_DAMAGE1);
+            }
+            if (damage) {
+                for (i = 0; i < 30; i++) {
+                    Vec3f pieceVel = { 0.0f, 0.0f, 0.0f };
+                    Vec3f pieceAccel = { 0.0f, -1.0f, 0.0f };
+                    Vec3f piecePos;
+
+                    pieceVel.x = Rand_CenteredFloat(6.0f);
+                    pieceVel.y = Rand_ZeroFloat(4.0f) + 6.0f;
+                    pieceVel.z = Rand_CenteredFloat(6.0f);
+
+                    piecePos.x = this->actor.focus.pos.x;
+                    piecePos.y = this->actor.focus.pos.y;
+                    piecePos.z = this->actor.focus.pos.z;
+
+                    BossFd2_SpawnSkullPiece(globalCtx, bossFd->effects, &piecePos, &pieceVel, &pieceAccel,
+                                            (s16)Rand_ZeroFloat(6.0f) + 10);
+                }
+            }
+        }
+    }
+}
+
+void BossFd2_UpdateFace(BossFd2* this, GlobalContext* globalCtx) {
+    f32 maxOpen;
+    f32 openRate;
+    s16 eyeStates[5] = { EYE_OPEN, EYE_HALF, EYE_CLOSED, EYE_CLOSED, EYE_HALF };
+
+    if (((this->work[FD2_VAR_TIMER] % 8) == 0) && (Rand_ZeroOne() < 0.3f)) {
+        this->work[FD2_BLINK_TIMER] = 4;
+    }
+    if ((this->actionFunc == BossFd2_Vulnerable) || (this->actionFunc == BossFd2_Damaged)) {
+        if (this->work[FD2_VAR_TIMER] & 0x10) {
+            this->eyeState = EYE_HALF;
+        } else {
+            this->eyeState = EYE_CLOSED;
+        }
+    } else {
+        this->eyeState = eyeStates[this->work[FD2_BLINK_TIMER]];
+    }
+    if (this->work[FD2_BLINK_TIMER] != 0) {
+        this->work[FD2_BLINK_TIMER]--;
+    }
+
```

Comment:
```suggestion
```

Reply:
fine place for a line break. imo every new block should be separated instead of all clumped together